src/annotator/util/buckets.js

```diff
@@ -88,161 +95,151 @@ export function findClosestOffscreenAnchor(anchors, direction) {
 }
 
 /**
- * Construct an Array of points representing the positional tops and bottoms
- * of current anchor highlights. Each anchor whose highlight(s)' bounding
- * box is onscreen will result in two entries in the `points` Array: one
- * for the top of the highlight box and one for the bottom
+ * Compute the AnchorPositions for the set of anchors provided, sorted
+ * by top offset
  *
  * @param {Anchor[]} anchors
- * @return {PositionPoints}
+ * @return {AnchorPosition[]}
  */
-export function constructPositionPoints(anchors) {
-  const aboveScreenAnchors = new Set();
-  const belowScreenAnchors = new Set();
-  const points = /** @type {PositionPoint[]} */ ([]);
+function getAnchorPositions(anchors) {
+  const anchorPositions = [];
 
-  for (let anchor of anchors) {
+  anchors.forEach(anchor => {
     if (!anchor.highlights?.length) {
-      continue;
+      return;
     }
+    const anchorBox = getBoundingClientRect(anchor.highlights);
+    anchorPositions.push({
+      top: anchorBox.top,
+      bottom: anchorBox.bottom,
+      anchor,
+    });
+  });
 
-    const rect = getBoundingClientRect(anchor.highlights);
-
-    if (rect.top < BUCKET_TOP_THRESHOLD) {
-      aboveScreenAnchors.add(anchor);
-    } else if (rect.top > window.innerHeight - BUCKET_NAV_SIZE) {
-      belowScreenAnchors.add(anchor);
-    } else {
-      // Add a point for the top of this anchor's highlight box
-      points.push([rect.top, 1, anchor]);
-      // Add a point for the bottom of this anchor's highlight box
-      points.push([rect.bottom, -1, anchor]);
-    }
-  }
-
-  // Sort onscreen points by pixel position, secondarily by position "type"
-  // (top or bottom of higlight box)
-  points.sort((a, b) => {
-    for (let i = 0; i < a.length; i++) {
-      if (a[i] < b[i]) {
-        return -1;
-      } else if (a[i] > b[i]) {
-        return 1;
-      }
+  // Now sort by top position
+  anchorPositions.sort((a, b) => {
+    if (a.top < b.top) {
+      return -1;
     }
-    return 0;
+    return 1;
   });
 
-  return {
-    above: Array.from(aboveScreenAnchors),
-    below: Array.from(belowScreenAnchors),
-    points,
-  };
+  return anchorPositions;
```

Comment:
You could write this function as one filter/map/sort chain. I'd be fine with it either way.